src/Connectivity/Connectivity.jl

```diff
@@ -0,0 +1,646 @@
+module Connectivity
+# Parts of this code were taken / derived from Graphs.jl. See LICENSE for
+# licensing details.
+using LightGraphs
+using LightGraphs.ShortestPaths
+using LightGraphs.Traversals
+using DataStructures: Queue, dequeue!, enqueue!
+using SimpleTraits
+import LightGraphs.Traversals: initfn!, newvisitfn!, postlevelfn!
+
+"""
+    connected_components!(label, g)
+
+Fill `label` with the `id` of the connected component in the undirected graph
+`g` to which it belongs. Return a vector representing the component assigned
+to each vertex. The component value is the smallest vertex ID in the component.
+
+### Performance
+This algorithm is linear in the number of edges of the graph.
+"""
+function connected_components!(label::AbstractVector, g::AbstractGraph{T}) where T
+    nvg = nv(g)
+    Q = Queue{T}()
+    @inbounds for u in vertices(g)
+        label[u] != zero(T) && continue
+        label[u] = u
+        enqueue!(Q, u)
+        while !isempty(Q)
+            src = dequeue!(Q)
+            for vertex in all_neighbors(g, src)
+                if label[vertex] == zero(T)
+                    enqueue!(Q, vertex)
+                    label[vertex] = u
+                end
+            end
+        end
+    end
+    return label
+end
+
+
+"""
+    components_dict(labels)
+
+Convert an array of labels to a map of component id to vertices, and return
+a map with each key corresponding to a given component id
+and each value containing the vertices associated with that component.
+"""
+function components_dict(labels::Vector{T}) where T <: Integer
+    d = Dict{T,Vector{T}}()
+    for (v, l) in enumerate(labels)
+        vec = get(d, l, Vector{T}())
+        push!(vec, v)
+        d[l] = vec
+    end
+    return d
+end
+
+"""
+    components(labels)
+
+Given a vector of component labels, return a vector of vectors representing the vertices associated
+with a given component id.
+"""
+function components(labels::Vector{T}) where T <: Integer
```

Comment:
```suggestion
function components(labels::AbstractVector{T}) where {T <: Integer}
```
Vector not required, also changed the formatting for consistency but totally optional